app/models/arclight/collection_downloads.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module Arclight
+  ##
+  # Model the Download links that can be configured (via YAML) for a collection
+  class CollectionDownloads
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def id
+      document.unitid
+    end
+
+    def files
+      data = config[id] || config['default']
+      return [] if data['disabled']
+
+      @files ||= data.map do |file_type, file_data|
+        Arclight::CollectionDownloads::File.new(type: file_type, data: file_data, document: document)
+      end.compact
+    end
+
+    def to_partial_path
+      'catalog/collection_downloads'
+    end
+
+    private
+
+    def config
```

Comment:
Are there future plans to make this config document-specific, or should it get extracted to the class level (or some other place?) so we're not continually re-parsing it?

Reply:
Decided to pull this up the class level.